catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -292,31 +292,28 @@ private String safeGetBody(Request req) {
       return ImmutableMap.of(RESP_MSG, "Could not create, no valid template specified");
     }
 
-    // Going to check to see if the metacard exists or not
-    Metacard oldMetacard = null;
+    // The UI should not send an ID during a PUT unless the metacard already exists
     String id = metacard.getId();
-    oldMetacard = getMetacardIfExistsOrNull(id);
-    if (oldMetacard != null) {
-      for (AttributeDescriptor descriptor :
-          oldMetacard.getMetacardType().getAttributeDescriptors()) {
-        Attribute metacardAttribute = metacard.getAttribute(descriptor.getName());
-        if (metacardAttribute == null || metacardAttribute.getValue() == null) {
-          continue;
+    if (id != null) {
```

Comment:
I would recommend inverting this predicate and returning early if `id == null` as that path has a much smaller amount of code to indent:

`if (id == null) {
    catalogFramework.create(new CreateRequestImpl(metacard));
    return ImmutableMap.of(RESP_MSG, "Successfully created");
}
...
//oldMetacard business
`

Reply:
Then I would need to code that part twice. If the old metacard is null I want to fallback to a create.